switcher.js

```diff
@@ -201,33 +228,141 @@ class Switcher {
     .then(data => this._handleMissingDefaultProfile(data));
   }
 
+  /**
+   * Parses the config file into an Array of profile objects,
+   * appends the config settings to the existing profile object
+   *
+   * @param {array} Profile objects - Existing array or profile objects
+   * @returns {Promise.<Array>} - Array of profile objects
+   * @reject {Error} - fs.readFile error
+   */
+  _getConfig(profileData) {
+    if (this.processConfig) {
+      return new Promise((resolve, reject) => {
+        fs.readFile(`${this.awsDir}/config`, 'utf8', (err, data) => {
+          if (err) {
+            return reject(err);
+          }
+
+          // Separate into lines
+          data = data.split('\n');
+
+          // Reduce the lines into an array of profile objects
+          data = data.reduce((prev, curr) => {
+            // Skip blank lines in the file
+            if (curr === '') {
+              return prev;
+            }
+
+            // This regex will match [.*] and extract interior characters as the second array item
+            // matches[0] = [AAAA]
+            // matches[1] = AAAA
+            const matches = curr.match(/\[(.*)\]/);
+
+            // If this is a profile name, push a new profile object to the array with the name
+            if (matches) {
+              // The config file user the pattern [profile profileName], need to strip the word profile
+              const [, name] = matches.map(match => match.replace(/profile/i, '').trim());
+              const existingProfile = prev.find(p =>
+                p.name !== undefined &&
+                Object.keys(p).length < 5
+              );
+              if (!existingProfile) {
+                prev.push({ name });
+              }
+              return prev;
+            }
+
+            /*
+            * At this point this line is neither blank nor a profile name
+            * Which means it's either an AWS profile ID or a configuration which are are formatted like:
+            *  output = OUTPUT
+            *  region = REGION
+            * With spaces in between the '=' and the key/value
+            */
+
+            // Split on '=' and then remove any spaces
+            // [0] === 'output'  || [0] === 'region'
+            // [1] === OUTPUT    || [1] === REGION
+            const [key, value] = curr.split('=').map(section => section.replace(' ', ''));
```

Comment:
If [this format](https://docs.aws.amazon.com/cli/latest/topic/config-vars.html) is possible

```
[profile testing]
aws_access_key_id = foo
aws_secret_access_key = bar
region = us-west-2
s3 =
  max_concurrent_requests=10
  max_queue_size=1000
```

then we're going to have issues here with storing values like this `s3` one. Need to figure out a way to handle those edge cases.